pysat/instruments/methods/general.py

```diff
@@ -208,3 +208,30 @@ def remove_leading_text(inst, target=None):
                 inst.meta[new_key] = new_meta
 
     return
+
+
+def filename_creator(value, format_str=None, start_date=None, stop_date=None):
+    """Creates filenames as needed to support use of generated pysat data sets
+
+    Parameters
+    ----------
+    value : slice
+        Datetime slice, see _instrument.py,
+        fname = self.files[date:(date + inc)]
+    format_str : str
+        File format template string (default=None)
+    start_date : datetime.datetime or None
+        First date supported (default=None)
+    stop_date: datetime.datetime or None
+        Last date supported (default=None)
+
+    Returns
+    -------
+    pandas.Series
+        Created filenames from format_str indexed by datetime
+
+    """
+
+    raise NotImplementedError()
```

Comment:
Maybe add an error message?  Also add a Raises section to the docstring (here and elsewhere in this class)